parse.go

```diff
@@ -273,10 +273,12 @@ func (c *Client) Do(req *http.Request, body, result interface{}) (*http.Response
 	}
 
 	res, err := c.RoundTrip(req)
+	if res.Body != nil {
```

Comment:
I don't think it can go up there as we still need the body to be readable if we make it to line 284 for the decoding.